qcodes/dataset/data_set.py

```diff
@@ -284,6 +287,18 @@ def __init__(self, path_to_db: str = None,
                 self._interdeps = InterDependencies_()
             self._metadata = get_metadata_from_run_id(self.conn, self.run_id)
 
+    @staticmethod
+    def _init_conn(conn, path_to_db):
+        if path_to_db is not None and conn is not None:
+            raise ValueError('Received BOTH conn and path_to_db. Please '
+                             'provide only one or the other.')
+        if conn is not None:
+            path_to_db = path_to_dbfile(conn)
+        else:
+            path_to_db = path_to_db or get_DB_location()
+        conn = conn or connect(path_to_db, get_DB_debug())
+        return conn, path_to_db
```

Comment:
wow, i never realized how complicatd this logic is. can we use this opportunity to make the code more readable? the `path_to_db or ...` parts distract me a lot.

Reply:
Sure I was guessing you were the clever guy who implemented this 😛